spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -227,4 +300,38 @@ private void applySelectWithFilter(Object[] parameters, Builder builder) {
 				: filters[0]);
 	}
 
+	private LimitedResult sliceQuery(Query query, Class<T> entityClass, Integer limit) {
```

Comment:
Maybe rename limit to conversionLimit, if you need it at all. Perhaps it can be pulled from the query itself.